packages/firebase_performance/ios/Classes/FirebasePerformancePlugin+Internal.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FirebasePerformancePlugin.h"
+
+@interface FLTFirebasePerformancePlugin (Internal)
++ (void)addMethodHandler:(NSNumber *_Nonnull)handle
+           methodHandler:(id<FlutterPlugin> _Nonnull)handler;
```

Comment:
I am not sure how I feel about using `id<FlutterPlugin>` as the type here. The empty `registerWithRegistrar` implementations feel like this is a hack that we wouldn't want to propagate to other plugins. These objects aren't really `FlutterPlugin`s, they are proxied classes. It might be a good idea to define a protocol makes this clear, perhaps something we can re-use for other plugins in the future.

Reply:
I was thinking the same thing when I was implementing the iOS side. On Android, they are called `MethodHandler`s and don't have static methods. 

I could create a separate protocol such as

```
@protocol MethodHandler
@required
- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result
@end


@interface FLTTrace : NSObject <MethodHandler>
@end
```